src/engine/runtime.js

```diff
@@ -439,6 +439,11 @@ Runtime.prototype.startHats = function (requestedHatOpcode,
     }
     var instance = this;
     var newThreads = [];
+
+    for (var opts in optMatchFields) {
+        optMatchFields[opts] = optMatchFields[opts].toUpperCase();
+    }
```

Comment:
I know where you are coming from, but if you do this within the function below then it will repeat the uppercase conversion once for every hat block rather than just once for the triggered event. If you want we can instead create a new array here to hold the uppercase arguments but I was just trying to save allocating more memory. Would you prefer that?